spec/unit/module_spec.rb

```diff
@@ -44,6 +43,31 @@
         expect(obj.send(:instance_variable_get, :'@expected_version')).to eq('8.0.0')
       end
     end
+
+    describe 'when the module is ostensibly on disk' do
+      before do
```

Comment:
What's the purpose of putting all this in a `before`? since there's only one test in this context..
I think if we just took this out of the `before` block, we could avoid the need for global variables (iiuc global variables should be a last resort?).

Reply:
The purpose is break up the test into more readable chunks; differentiating the setup of what ostensibly looks like a module on disk from the rest of the test improves the readability.

Reply:
hmm I do agree that this is very readable.
Am I wrong that global variables should be avoided whenever possible?
Would a `# Setup` comment or something accomplish the same thing? Or do we think it's likely we'd add additional test cases to this, in which case a `before` would make sense?

Reply:
I don't think there's a hard and fast answer of when to avoid using rspec instance variables; the biggest thing to look out for are tests that might modify them if instance variables are passed for whole context `before{:context)`.
I have no idea where we might need test coverage next, but the way it is here makes it clear that this is a place to put tests that require faking a module installed on disk, or at least provides the template for copying that strategy.